pkg/rancher-components/src/components/Card/Card.vue

```diff
@@ -56,32 +57,17 @@ export default defineComponent({
       default: false,
     },
   },
-  data() {
-    return { focusTrapInstance: {} as FocusTrap };
-  },
-  mounted() {
-    if (this.triggerFocusTrap) {
-      this.focusTrapInstance = createFocusTrap(this.$refs.cardContainer as HTMLElement, {
-        escapeDeactivates: true,
-        allowOutsideClick: true,
-      });
-
-      this.$nextTick(() => {
-        this.focusTrapInstance.activate();
-      });
+  setup(props) {
+    if (props.triggerFocusTrap) {
+      basicSetupFocusTrap('#focus-trap-card-container');
     }
-  },
-  beforeUnmount() {
-    if (this.focusTrapInstance && this.triggerFocusTrap) {
-      this.focusTrapInstance.deactivate();
-    }
-  },
+  }
 });
 </script>
 
 <template>
   <div
-    ref="cardContainer"
+    id="focus-trap-card-container"
```

Comment:
Is there any reason we can't stick with the ref? I think that it is preferred over an `id` when working with component instances.

Reply:
I've expanded all of the composable so that it would use an `HTMLElement` as well as selector (In theory it should work, according to https://github.com/focus-trap/focus-trap).

In this particular case, the actual usage of `useBasicSetupFocusTrap` depends on the `setup` function on which, after multiple tries, I could never get the `ref` to work 😬  . Let me know if you wanna take a stab at this

Reply:
It looks like it _could_ work if we did something like this in the composable:

```js
export function basicSetupFocusTrap(containerRef: string) {
  let focusTrapInstance = {} as FocusTrap;
  const modalRef = useTemplateRef<HTMLElement>(containerRef);

  onMounted(() => {
    if (!modalRef.value) {
      return;
    }

    focusTrapInstance = createFocusTrap(modalRef.value, {
   ...
```

But this is completely optional if we have a universally unique ID that we can reference instead.

Reply:
Created https://github.com/rancher/dashboard/issues/13344 to track this. Assigned it to `2.12` milestone as it's definitely not a priority